addons/stock/migrations/11.0.1.1/post-migration.py

```diff
@@ -59,32 +59,22 @@ def product_assign_responsible(env):
 
 @openupgrade.logging()
 def create_specific_procurement_rules_from_globals(env):
-    """Create one record per route for the global rules found in previous
-    version.
-    """
+    """Update global rules by linking them to a global route."""
+    warehouses = env["stock.warehouse"].with_context(active_test=False).search(
+        [])
+    global_route = env["stock.location.route"].create({
```

Comment:
I think this route gets created when there are no dangling rules or paths. Maybe you can fetch those first, and only create this route if necessary. You can refactor out the SQL queries by writing pull_ids and push_ids directly at creation time.

Reply:
I think it's done! :)